lib/async_hooks.js

```diff
@@ -342,24 +345,28 @@ function emitInitS(asyncId, type, triggerAsyncId, resource) {
   }
 }
 
-
-function emitBeforeN(asyncId) {
-  processing_hook = true;
-  // Use a single try/catch for all hook to avoid setting up one per iteration.
-  try {
-    for (var i = 0; i < active_hooks_array.length; i++) {
-      if (typeof active_hooks_array[i][before_symbol] === 'function') {
-        active_hooks_array[i][before_symbol](asyncId);
+function hookFactory(symbol) {
+  // Called from native. The asyncId stack handling is taken care of there
+  // before this is called.
+  return function(asyncId) {
```

Comment:
The stack trace was changed so I now explicitly set the name and it's now looking good again.

```
Error
    at AsyncHook.before (repl:2:26)
    at emitBeforeN (async_hooks.js:358:40)
    at emitBeforeS (async_hooks.js:393:3)
    at nextTickEmitBefore (internal/process/next_tick.js:158:7)
    at process._tickDomainCallback (internal/process/next_tick.js:230:9)
```